### PR TITLE
[docs] Polish pickers migration docs

### PR DESCRIPTION
**Pull request by LukasTy**
- Add "Rename `components` to `slots`" section
- Fix typos
- Fix `codemod` script examples message to contain a valid example

### CODE REVIEW

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 1]
```diff
@@ -853,3 +853,16 @@ Component name changes are also reflected in `themeAugmentation`:
    },
  });
 ```
+
+## Rename `components` to `slots` (optional)
+
+The `components` and `componentsProps` props is going to be renamed to `slots` and `slotProps` props respectively.
```
Comment (flaviendelangle) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:859]: Maybe

```suggestion
The `components` and `componentsProps` props is being renamed to `slots` and `slotProps` props respectively.
```

**Review by flaviendelangle — APPROVED** [review]

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 2]
```diff
@@ -853,3 +853,16 @@ Component name changes are also reflected in `themeAugmentation`:
...
+This is a slow an ongoing effort between the different MUI libraries, that's why we chose to proceed with deprecating the existing `components` props as well as adding the new `slots` props.
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:860]:
```suggestion
This is a slow and ongoing effort between the different MUI libraries.
To smooth the transition, pickers support both the `components` props which are deprecated, and the new `slots` props.
```

Comment (LukasTy) [reply docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:860]: 🙈 🤦

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 3]
```diff
@@ -853,3 +853,17 @@ Component name changes are also reflected in `themeAugmentation`:
    },
  });
 ```
+
+## Rename `components` to `slots` (optional)
+
+The `components` and `componentsProps` props is being renamed to `slots` and `slotProps` props respectively.
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:859]:
```suggestion
The `components` and `componentsProps` props are being renamed to `slots` and `slotProps` props respectively.
```

**Pull request merged**